src/go/rpk/pkg/adminapi/api_transform.go

```diff
@@ -33,6 +37,31 @@ func (a *AdminAPI) DeployWasmTransform(ctx context.Context, t TransformMetadata,
 	return a.sendToLeader(ctx, http.MethodPost, baseTransformEndpoint+"deploy", body, nil)
 }
 
+func validateWasmBinary(file io.Reader) error {
+	// Define the magic bytes for WebAssembly binary module
+	wasmMagicBytes := []byte{0x00, 0x61, 0x73, 0x6D}
+
+	// Not using bufio.Reader because it feels weird to spawn one just to peek
+	buf := make([]byte, 4)
+	if _, err := io.ReadFull(file, buf); err != nil {
+		return fmt.Errorf("error reading file: %w", err)
+	}
+
+	if !bytes.Equal(buf, wasmMagicBytes) {
+		return fmt.Errorf("invalid file format: missing WebAssembly magic bytes")
+	}
+
+	// Reset the file pointer to the beginning of the file
+	seeker, ok := file.(io.Seeker)
```

Comment:
This seems like a dangerous abstraction. Since all the callers of this are working with a full bytes slice can we just update the interface to take a full byte slice and then create the BytesReader when we create the MultiReader?

Reply:
@rockwotj ok, reworked to just pass around []byte